src/test/codegen/array-equality.rs

```diff
@@ -0,0 +1,45 @@
+// compile-flags: -O
+// only-x86_64
+
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @array_eq_value
+#[no_mangle]
+pub fn array_eq_value(a: [u16; 6], b: [u16; 6]) -> bool {
```

Comment:
`[u16; 9]` would be a good test as well--it's big enough (>2*usize) that it'll be passed by reference, but small enough for this PR to load it as an `i144`, which is an unusual type (e.g. https://godbolt.org/z/3PGbcsKEr)

Reply:
Thanks!  That's a great example.  (Also prompted me to check that for the `&[u16; 6]` case too, but it seems to generate the same ASM either way <https://godbolt.org/z/e6bYK9qne>)

In response I've changed the intrinsic implementation for LLVM to look at the ABI, not just the size.  I think that fits as a better threshold, since most of the advantage here comes from avoiding allocas, but if it's big enough that it's getting passed by pointer anyway, that's less likely to be applicable.